src/main/kotlin/calculator/Operator.kt

```diff
@@ -0,0 +1,28 @@
+package calculator
+
+import java.util.Arrays
+
+enum class Operator(
+    private val symbol: String,
+    private val function: (a: Int, b: Int) -> Int
+) {
+
+    ADDITION("+", { a, b -> a + b }),
+    SUBTRACTION("-", { a, b -> a - b }),
+    MULTIPLICATION("*", { a, b -> a * b }),
+    DIVISION("/", { a, b -> a / b }),
+    ;
+
+    fun operate(a: Int, b: Int): Int {
+        return function.invoke(a, b)
+    }
+
+    companion object {
+        fun of(symbol: String): Operator {
+            return Arrays.stream(values())
+                .filter { operator -> operator.symbol == symbol } // 코틀린에서는 equals 비교가 ==
+                .findFirst()
+                .orElseThrow { IllegalArgumentException("지원하지 않는 연산자 입니다. ($symbol)") }
+        }
```

Comment:
```suggestion
        fun of(symbol: String): Operator = values().first { it.symbol == symbol }
```
java 라이브러리 대신에 kotlin 고유의 collection 함수를 써보시는걸 추천드립니다! 
참고로 kotlin 의 first 함수를 사용하면 간결하게 표현이 가능합니다!